src/Mvc/Mvc.ViewFeatures/src/RazorComponents/Prerrendering/IPrerrenderer.cs

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Server
+{
+    public interface IComponentPrerrenderer
```

Comment:
Nit: XML docs

Reply:
Also, is there any way this could be `internal`? I'm not sure what the use case would be for customers getting the `IEnumerable<string>` directly themselves.

Reply:
Needs to be public as is the main abstraction that both MVC and Components.Server share. We can change the IEnumerable<string> design to support alternatives.

It's as simple as PrerrenderResult.WriteTo(stream/pipe)